catchup/universalFetcher.go

```diff
@@ -195,7 +198,11 @@ func (w *wsFetcherClient) requestBlock(ctx context.Context, round basics.Round)
 // set max fetcher size to 10MB, this is enough to fit the block and certificate
 const fetcherMaxBlockBytes = 10 << 20
 
-var errNoBlockForRound = errors.New("No block available for given round")
+type noBlockForRoundError struct {
+	latest, round basics.Round
+}
+
+func (noBlockForRoundError) Error() string { return "no block available for given round" }
```

Comment:
If you don't put `latest` or `round` in the error string, what's the point of recording these fields? Is consuming these supposed to happen in a different PR?

Reply:
yes, planning to put start using it to do smarted catchup in follower nodes